modin/pandas/test/internals/test_benchmark_mode.py

```diff
@@ -11,21 +11,55 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
-# test BenchmarkMode == True
+import unittest.mock as mock
 
-from contextlib import nullcontext
 import modin.pandas as pd
 from modin.pandas.test.utils import test_data_values
-from modin.config import BenchmarkMode, StorageFormat
-from modin.test.test_utils import warns_that_defaulting_to_pandas
+from modin.config import BenchmarkMode, Engine
 
 
-def test_syncronous_mode():
+engine = Engine.get()
+
+# We have to explicitly mock subclass implementations of wait_partitions.
+if engine == "Ray":
+    wait_method = (
+        "modin.core.execution.ray.implementations.pandas_on_ray."
+        + "partitioning.partition_manager."
+        + "PandasOnRayDataframePartitionManager.wait_partitions"
+    )
+elif engine == "Dask":
+    wait_method = (
+        "modin.core.execution.dask.implementations."
+        + "pandas_on_dask.partitioning.partition_manager."
+        + "PandasOnDaskDataframePartitionManager.wait_partitions"
+    )
+else:
+    wait_method = (
+        "modin.core.dataframe.pandas.partitioning."
+        + "partition_manager.PandasDataframePartitionManager.wait_partitions"
+    )
+
+
+def test_from_environment_variable():
     assert BenchmarkMode.get()
-    # On Omnisci storage, transpose() defaults to Pandas.
-    with (
-        warns_that_defaulting_to_pandas()
-        if StorageFormat.get() == "Omnisci"
-        else nullcontext()
-    ):
+    with mock.patch(wait_method) as wait:
         pd.DataFrame(test_data_values[0]).mean()
+
+    wait.assert_called()
+
+
+def test_turn_off():
```

Comment:
I would combine both of the tests below into one.

Reply:
I very much prefer that tests test only one behavior (see e.g. [here](https://stackoverflow.com/a/61868/17554722) for why). I think many Modin unit tests are too long. When a single test case has many assertions within it that may fail, it's hard to tell from the unit test report what exactly is going wrong. Ideally we should be able to use the name of each failed test to identify one incorrect behavior.

Reply:
Ok, let's leave as is.

Reply:
That said, we can do two tests: 1) Test for the class config setting, 2) Test for the env var setting. I think it would be okay.